File: Projects/01/rect_area.rb
#! /usr/bin/ruby
# rect_area.rb computes the area of a rect given its length and width
# 
# Input: the length and width of a rect
# Precondition: the length and width is not negative
# Output: the area of the rect.
#
# Begun by: Prof. Adams, for CS 214 at Calvin College.
# Completed by:
# Date:
###############################################################

# function rectArea returns a rect's area, given its length and width
# Parameters: length and width, a number
# Precondition: length and width > 0.
# Returns: the area of a rect whose length and width is length and width.
def rectArea(length, width)
    length * width 
end

if __FILE__ == $0
   puts "To compute the area of a rect,"
   print " enter its length: "
   length = gets.chomp.to_f
   print " enter its height: "
   height = gets.chomp.to_f
   print "The rect's area is: "
   puts rectArea(length, height)
end

File: Projects/01/rect_area.clj
;;;; rect_area.clj calculates the area of a rect.
;;;;
;;;; Input: The radius of a rect.
;;;; Output: The area of that rect.
;;;;
;;;; Usage: clojure -m rect_area
;;;;
;;;; Begun by: Prof. Adams, for CS 214 at Calvin College.
;;;; Completed by:
;;;; Date:
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns rect_area) ; namespace function names the program

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; Function rectArea() computes the area of a rect.
;;; Receive: itsLength, itsHeight, a number.
;;; Precondition: itsLength, itsHeight >= 0.0.
;;; Return: the area of the corresponding rect.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(defn rectArea [itsLength itsHeight]
  (* itsLength itsHeight 1.0)  
)
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; Function -main is where execution begins
;;; Input: the height and length of a rect.
;;; Output: the area of that rect.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn -main []
  (println "\nTo compute the area of a rect,")
  (print   " enter its height: ") (flush)
  (let
    [ height (read) ]
    (assert (>= height 0) "-main: height must be positive")

    (print   " enter its length: ") (flush)
    (let
     [ length (read) ]


     (assert (>= length 0) "-main: length must be positive")
     (printf "\nThe area is %f\n\n" (rectArea length height))

     (print "\nThe area is ")
     (print (rectArea length height))
     (print "\n\n")

    (printf "\nThe area is %.15f\n\n" (rectArea length height))
    )
  )
)